blockchains/erc20/lib/timestamps_cache.js

```diff
@@ -1,45 +1,54 @@
 'use strict';
-const DATA_MISSING = -1;
 
 class TimestampsCache {
-  constructor() {
+  constructor(ethClient, fromBlock, toBlock) {
     this.timestampStore = {};
-  }
+    this.rangeSize = toBlock - fromBlock + 1;
+
+    const blockRequests = [];
+    for (let i = fromBlock; i <= toBlock; i++) {
```

Comment:
One of the two main features of this PR. In this TimestampsCache constructor we would send the batch RPC request to get all the blocks for the range we are currently fetching. What we did before was do a separate request once we need a new block, this would result in N requests if we are fetching a range of N blocks.

Reply:
![image](https://github.com/santiment/san-chain-exporter/assets/739463/cebfc815-fb59-497d-b48c-86733729987f)

Reply:
I thought it's a classic we all like, no? :) You would suggest descriptive names like 'counter'?